src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,173 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+
+    T item;
+    MyLinkedList<T> next;
+    MyLinkedList<T> prev;
+    transient MyLinkedList<T> first;
+    transient MyLinkedList<T> last;
+    MyLinkedList() {
+    }
+    MyLinkedList(MyLinkedList<T> prev, T element, MyLinkedList<T> next) {
+        this.item = element;
+        this.next = next;
+        this.prev = prev;
+        //this.last = this;
+        //this.first = this;
+
+    }
     @Override
     public void add(T value) {
+        if (isEmpty()) {
+            MyLinkedList newNode = new MyLinkedList(null, value, null);
+            first = newNode;
+            last = newNode;
+        } else {
+            MyLinkedList newNode = new MyLinkedList(last, value, null);
+            last.next = newNode;
+            last = newNode;
+        }
     }
 
     @Override
-    public void add(T value, int index) {
+    public void add(T value, int index) throws IndexOutOfBoundsException  {
+        if ((index > size()) || (index<0)) {
+            throw new IndexOutOfBoundsException("Index dosn't much");
+        }
+        if (index == size()) {
+            add(value);
+            return;
+        }
+        if (index == 0) {
+            MyLinkedList newNode = new MyLinkedList(null, value, first);
+            first.prev = newNode;
+            first = newNode;
+            return;
+        }
+        int i = 0;
+        MyLinkedList node = first;
+        while (i != index){
+            node = node.next;
+            i++;
+        }
```

Comment:
Pls recheck checklist
https://github.com/mate-academy/jv-linked-list/blob/master/checklist.md#dont-iterate-your-nodes-from-the-head-to-the-tail-or-vice-versa